src/mscorlib/src/System/Decimal.cs

```diff
@@ -378,7 +378,7 @@ public int CompareTo(Object value)
             if (value == null)
                 return 1;
             if (!(value is Decimal))
-                throw new ArgumentException(Environment.GetResourceString("Arg_MustBeDecimal"));
+                throw new ArgumentException(Environment.GetResourceString("Arg_MustBeDecimal"), nameof(value));
```

Comment:
Perhaps I revert my change in this PR and open a separate one addressing all `CompareTo` methods?